### ISSUE TITLE
savenochange should be triggered when changing from null or undefined to false

### ISSUE THREAD

**Issue by richard-flosi** [vitalets/x-editable#634]
In `jquery-editable-poshytip.js` I see the following:
```
            //if value not changed --> trigger 'nochange' event and return
            /*jslint eqeq: true*/
            if (!this.options.savenochange && this.input.value2str(newValue) == this.input.value2str(this.value)) {
            /*jslint eqeq: false*/                
                /**        
                Fired when value not changed but form is submitted. Requires savenochange = false.
                @event nochange 
                @param {Object} event event object
                **/                    
                this.$div.triggerHandler('nochange');            
                return;
            } 
```

I think the issue is in `value2str`.

I have created a boolean select (dropdown) with `null`, `true`, and `false` options. Trying to switch from `null` to `false` does not trigger a change.

To work around this for now I've setup this field to set `savenochange` to true, but I think this should be updated to evaluate a change from `null` to `false` as a change instead of `nochange`.

**Comment by tofutim** [comment]
We should write a test for this.

**Comment by tofutim** [comment]
Yes the issue is in value2str which was not even turning anything to a str. In dev, I have made value2str return String(value).

**Comment by tofutim** [comment]
The fix for #834 should also fix this. It should now compare 'null' to 'false' and do the right thing.
